CT-2917 update the case details page

njpreddy opened the pull request:
## Description
CT-2917 Update the case details section

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
Case 1 - Third party requests data to be sent to the data subject

![Case1](https://user-images.githubusercontent.com/6988369/87773526-32ca1c80-c81b-11ea-87ca-c7400a94f6b9.png)

Case 2 - Third party requests data for themselves

![Case2](https://user-images.githubusercontent.com/6988369/87773543-39f12a80-c81b-11ea-9e07-acb599a19a5b.png)

Case 3 - Data subject requesting their own data

![Case3](https://user-images.githubusercontent.com/6988369/87773569-42e1fc00-c81b-11ea-8423-93129061a20c.png)

Case 4 - Data subject requesting data to be sent to third party

![Case4](https://user-images.githubusercontent.com/6988369/87773601-4d03fa80-c81b-11ea-8ea4-3204c1764dc3.png)

### Related JIRA tickets
Jira ticket CT-2917
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2917

### Deployment

### Manual testing instructions
Create the following cases with the appropriate steps:

<img width="1033" alt="87925944-ebd96280-ca78-11ea-877f-41163baaeca1" src="https://user-images.githubusercontent.com/6988369/88053161-46df8800-cb53-11ea-8ec1-063f6e242476.png">

njpreddy committed:
CT-2917 Update the case details section

*Replace hard coded change link name with locale file entry.

njpreddy committed:
CT-2917 Update the case details section

*Add third party address field
*Move third party relationship field below address
*Rename 'Requester' field to 'Representative name'
*Rename 'Location of data subject' field to 'Location'
*Sort locale common.case keys alphabetically
*Delete duplicate key in locale common.case
*Check recipient type to output correct 'where should data be sent' label
*Unless requester is 'requester_recipient' display third party details.

njpreddy committed:
CT-2917

*Add get recipient label helper method.
*Fix when third party details are displayed condition.

njpreddy committed:
Merge branch 'master' into ct-2917-update-the-case-details-page

njpreddy committed:
CT-2917 Fix locale key typo.

willmcb reviewed (commented):
Looks good 👍 - I have a few suggestions - let me know if you'd like to chat about them

njpreddy committed:
CT-2917 extract recipient labels into locale file

njpreddy committed:
CT-2917

*Move relationship field to top of recipient details form
  in line with requester details form

*Chunk case details fields together into sections

njpreddy committed:
CT-2917 Fix case details page tests

Polsonby reviewed (commented):
Looks great! I wasn't sure about nuking all the "one liner" view partials for each field, since that's how the other correspondence types are displayed but I really like the way the offender sar view files have ended up when you look at the files side by side in the folder, I think we're in a nice place with this PR :) 

There are a few tiny things we could potentially consider tweaking, but not sure I'd hold up merging for them! 

Happy to approve either way - just wondering if you might be able to make one change to request dated as [per comment above](https://github.com/ministryofjustice/correspondence_tool_staff/pull/1205#discussion_r458413703) though?

njpreddy committed:
CT-2917

*Fix crash on optional partial if date field is nil.

njpreddy committed:
CT-2917

*Simplifying 'if' conditions for partial display.

Polsonby reviewed (approved):
Nice work :)
